python_scripts/trees_hyperparameters.py

```diff
@@ -28,73 +28,39 @@
 # %% [markdown]
 # ## Create helper functions
 #
-# We will create two functions that will:
-#
-# * fit a decision tree on some training data;
-# * show the decision function of the model.
+# We will create some helper functions to plot the data samples as well as the
+# decision boundary for classification and the regression line for regression.
 
 # %%
 import numpy as np
 import matplotlib.pyplot as plt
 import seaborn as sns
+from helpers.plotting import DecisionBoundaryDisplay
 
 
-def plot_classification(model, X, y, ax=None):
-    from sklearn.preprocessing import LabelEncoder
-    model.fit(X, y)
-
-    range_features = {
-        feature_name: (X[feature_name].min() - 1, X[feature_name].max() + 1)
-        for feature_name in X.columns
-    }
-    feature_names = list(range_features.keys())
-    # create a grid to evaluate all possible samples
-    plot_step = 0.02
-    xx, yy = np.meshgrid(
-        np.arange(*range_features[feature_names[0]], plot_step),
-        np.arange(*range_features[feature_names[1]], plot_step),
-    )
-    grid = pd.DataFrame(
-        np.c_[xx.ravel(), yy.ravel()],
-        columns=[feature_names[0], feature_names[1]],
-    )
-
-    # compute the associated prediction
-    Z = model.predict(grid)
-    Z = LabelEncoder().fit_transform(Z)
-    Z = Z.reshape(xx.shape)
-
-    # make the plot of the boundary and the data samples
-    if ax is None:
-        _, ax = plt.subplots()
-    ax.contourf(xx, yy, Z, alpha=0.4, cmap="RdBu")
-    if y.nunique() == 3:
-        palette = ["tab:red", "tab:blue", "black"]
-    else:
+def plot_classification(model, data, feature_names, target_names):
```

Comment:
Maybe you could rename the function to `fit_and_plot_classification` and actually do the call to fit in the function?

Reply:
Since we always filtering the columns and target, I think this is better to include it in the function.